caikit/runtime/http_server.py

```diff
@@ -290,6 +385,7 @@ async def _handler(
             loop = asyncio.get_running_loop()
 
             request_params = self._get_request_params(rpc, request)
+            self.build_request_params_dict(request_params)
```

Comment:
nit: This function says it returns a dict, but it looks like it's also modifying the input dict in-place

Quick questions:
- Why is this here on the inference endpoints, is it a bugfix?
- I see this is only called right after `self._get_request_params`, should it just be wrapped into that?

Reply:
Yeah could be combined I guess!

Reply:
inference endpoints also need pydantic models to be converted to our DM objects right?

Reply:
I suppose that actually wasn't happening before and the equivalent pydantic objects were just being passed along?
In that case duck-typing would be fine, but any code that a module author wrote that type-checked against the data model type would fail. Is that why there's now a `SampleInputType` type check in one of the test modules?

Reply:
Exactly.

Reply:
Sweet, solid upgrade 👍